src/gcodePlanner.cpp

```diff
@@ -409,15 +409,23 @@ void GCodePlanner::addPolygonsByOptimizer(Polygons& polygons, GCodePathConfig* c
     {
         return;
     }
-    PathOrderOptimizer orderOptimizer(lastPosition, z_seam_pos, z_seam_type);
-    for (unsigned int poly_idx = 0; poly_idx < polygons.size(); poly_idx++)
+    if (spiralize)
     {
-        orderOptimizer.addPolygon(polygons[poly_idx]);
+        // only interested in the outer perimeter
+        addPolygon(polygons[0], 0, config, wall_overlap_computation, wall_0_wipe_dist, spiralize);
```

Comment:
While it is an interesting thing to take only the outer polygon in spiralize mode, this is not how spiralize is intended.
Think about a C shape which is gradually closing up into an O when going upward. Your change would entail that at some point suddenly half of the lines aren't printed any more.

Moreover, the first polygon in a Polygons object isn't guaranteed to be the outer polygon. That is only so for a PolygonsPart.

I would like you to revert this change.